bigframes/core/compile/single_column.py

```diff
@@ -64,6 +64,8 @@ def join_by_column_ordered(
             .equals(right._get_ibis_column(rcol).name("index"))
             for lcol, rcol in join.conditions
         )
+        and not left._predicates
```

Comment:
This would likely be a regression with regards to our SQL size. We want to simulate a join in this case. I'll take a look at `join_by_row_identity_ordered` and `join_by_row_identity_unordered` to see if there has been a regression there.

Reply:
any update on this?

Reply:
Hi @DLT1412 Sorry for the wait. I was OOO for a couple weeks to get married. I hope to look at this more deeply later this week. Thanks again for raising this issue.

Reply:
I'm able to reproduce this issue locally with inner, outer, and left join. Will look more tomorrow.